### ISSUE TITLE
[A11y] General - My Account

### ISSUE THREAD

**Issue by jechols** [OregonDigital/OD2#1805]
### Operating System

Windows

### Browser

Edge

### Affected page

Homepage

### Admin user?

Yes

### Issue Description

The "My Account" dropdown implements some of the menu pattern, but not correctly and not completely.

### Suggested Fix

A disclosure is fine here.  A menu may be used as well, but a "navigation" disclosure probably makes the most sense since it's just a simple four-item list of links and is easier to implement for all the keyboard controls.

W3C info:
- Disclosures: https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure
- Disclosure example for navigation: https://www.w3.org/TR/wai-aria-practices-1.1/examples/disclosure/disclosure-navigation.html
- Menus: https://www.w3.org/TR/wai-aria-practices-1.1/#menu
- Menu example for navigation: https://www.w3.org/TR/wai-aria-practices-1.1/examples/menubar/menubar-1/menubar-1.html

If we want this to be a disclosure, we need to fix a few things:

- The dropdown doesn't close when you tab off the disclosure.
- When focus is on the button, the spacebar isn't closing the disclosure.  It seems to intermittently do so, but most of the time it puts focus on the first item in the list.  Since it closes with ESC, this might not be a hard failure, I'm not sure.
- Obviously the role would need to change since right now it's pretending to be a menu.

If we want this to be using the menu pattern, it's a bit more work to remediate:

- When the button is focused, enter and space should move focus to the first item in the list
- Home and End keys have to move to the top/bottom of the menu list unless arrow keys wrap, which they currently do not
- The menu should act as a single focusable widget as far as the "tab" key is concerned - tabbing should close the menu and move to the next focusable widget
- The menu needs to automatically close once it doesn't have focus
- When a link is focused, the spacebar has to activate it

**Comment by CGillen** [comment]
I did manage to find a solution, it wasn't ultimately terribly involved but it took figuring out some JS events and thinking a bit more outside the scope of the menu. Leaving the context of the menu should close it now though =]

**Comment by jechols** [comment]
@CGillen I think this is almost correct, but there is an invalid aria attribute on here that seems to be causing NVDA some confusion. The HTML is:

```html
<a id="user-dropdown" class="btn btn-outline" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" aria-controls="user-util-links" href="/dashboard/profiles/admin@example-dot-org?locale=en">
        <span class="fa fa-user" aria-hidden="true"></span>
        <span class="hidden-xs">&nbsp;My Account</span>
        <span class="caret"></span>
</a>
```

The problem (I believe) is the `aria-haspopup`. The [W3C aria-haspopup section](https://w3c.github.io/aria/#aria-haspopup) states:

> A popup element usually appears as a block of content that is on top of other content. Authors MUST ensure that the role of the element that serves as the container for the popup content is menu, listbox, tree, grid, or dialog, and that the value of aria-haspopup matches the role of the popup container.

I believe that means a disclosure button should not have this attribute since the popup content isn't any of those roles.
